### PR TITLE
Ignore jars that have only packages from a single vendor prefix (unle…

### PR DESCRIPTION
**Pull request by jsight**
…ss explicitly included in analysis)

### CODE REVIEW

File: rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/reporting/rules/CreateJavaIgnoredFilesReportRuleProvider.java [hunk 1]
```diff
@@ -35,6 +35,12 @@
 {
     public static final String TITLE = "Ignored Files";
     public static final String TEMPLATE_REPORT = "/reports/templates/ignored_files.ftl";
+    public static final String DESCRIPTION = "This report lists the files which were found in the application,\n" +
+            "                    but based on certain rules and the Windup configuration, both built-in and local,\n" +
+            "                    they were not processed.\n" +
+            "                    See the <code>--userIgnorePath</code> Windup option in\n" +
+            "                    <a href=\"http://windup.github.io/windup/docs/latest/html/WindupUserGuide.html#command-line-arguments\"\n" +
+            "                       >Windup User Guide</a>.";
```
Comment (mareknovotny) [inline rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/reporting/rules/CreateJavaIgnoredFilesReportRuleProvider.java]: we might need to add specific anchors for option argument in WIKI doc to point directly into it.

**Comment by mareknovotny** [comment]
I can see there are not files grouped by name and due this the table for known jars contains multiple lines with the same "duplicated" archive names. The Jar dependencies report has got these groupped somehow even not correctly too.
screenshot from 2016-04-18 12-25-56

**Comment by mareknovotny** [comment]
Is the above wanted?

**Comment by jsight** [comment]
I wouldn't exactly call it "wanted". However, I think it is a separate issue as this report existed before this PR as well.

File: rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/scan/operation/packagemapping/PackageNameMapping.java [hunk 2]
```diff
@@ -68,6 +75,76 @@ public Rule toOrganization(String organization)
...
+
+            if (entry.isDirectory() || !StringUtils.endsWith(entryName, ".class"))
+                continue;
+
+            String classname = PathUtil.classFilePathToClassname(entryName);
+            // if the package isn't current "known", try to match against known packages for this entry.
+            if (!exclusivelyKnown)
+            {
+                organization = getOrganizationForPackage(event, classname);
+                if (organization != null)
```
Comment (OndraZizka) [inline rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/scan/operation/packagemapping/PackageNameMapping.java]: Where do we get this mapping?

File: rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/scan/operation/packagemapping/PackageNameMapping.java [hunk 3]
```diff
@@ -68,6 +75,76 @@ public Rule toOrganization(String organization)
...
+        String organization = null;
+        Enumeration<?> e = archive.entries();
+
+        // go through all entries...
+        ZipEntry entry;
+        while (e.hasMoreElements())
+        {
+            entry = (ZipEntry) e.nextElement();
+            String entryName = entry.getName();
+
```
Comment (OndraZizka) [inline rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/scan/operation/packagemapping/PackageNameMapping.java]: Shouldn't we put the packages into graph? It's not that much information, and I'm going to need that for Mavenization; should I scan the zips again?

File: rules-java/api/src/main/resources/reports/templates/ignored_files.ftl [hunk 4]
```diff
@@ -35,23 +35,25 @@
...
-            </tr>
-            </#list>
-        </table>
-    </div>
+                <#list reportModel.fileRegexes.iterator() as regex>
+                <tr>
+                    <td> <#if regex.regex?has_content> ${regex.regex} </#if> </td>
+                    <td> <#if regex.compilationError?has_content> ${regex.compilationError}
+                         <#else> OK
+                         </#if> </td>
```
Comment (OndraZizka) [inline rules-java/api/src/main/resources/reports/templates/ignored_files.ftl]: How about `${regex.compilationError!"OK"}` ?

File: rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/scan/operation/packagemapping/PackageNameMapping.java (hunk 2 above)
Comment (jsight) [inline rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/scan/operation/packagemapping/PackageNameMapping.java]: https://github.com/windup/windup-rulesets/blob/master/rules-reviewed/filemappings/default-package-to-organization-names.windup.xml

File: rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/scan/operation/packagemapping/PackageNameMapping.java (hunk 3 above)
Comment (jsight) [inline rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/scan/operation/packagemapping/PackageNameMapping.java]: We do have package names in the graph. I'm also not sure why you would need this for mavenization?

File: rules-java/api/src/main/resources/reports/templates/ignored_files.ftl (hunk 4 above)
Comment (jsight) [inline rules-java/api/src/main/resources/reports/templates/ignored_files.ftl]: That probably would have been better. I haven't changed any of this code, though.

**Comment by jsight** [comment]
@OndraZizka Is this one ready to merge?

File: rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/scan/operation/packagemapping/PackageNameMapping.java (hunk 3 above)
Comment (OndraZizka) [inline rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/scan/operation/packagemapping/PackageNameMapping.java]: For the record, We've discussed that on video chat. I need a bit different kind of info about packages.

File: rules-java/api/src/main/resources/reports/templates/ignored_files.ftl (hunk 4 above)
Comment (OndraZizka) [inline rules-java/api/src/main/resources/reports/templates/ignored_files.ftl]: Ok

File: rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/scan/operation/packagemapping/PackageNameMapping.java (hunk 3 above)
Comment (jsight) [inline rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/scan/operation/packagemapping/PackageNameMapping.java]: +1 - You sound like you need it at a different level of execution for a different purpose.

File: rules-java/api/src/main/resources/reports/templates/ignored_files.ftl (hunk 4 above)
Comment (jsight) [inline rules-java/api/src/main/resources/reports/templates/ignored_files.ftl]: I also went ahead and changed it to your preferred format. That is clearly better. :)

**Pull request merged**